packages/appkit/src/universal-adapter/client.ts

```diff
@@ -246,12 +247,35 @@ export class UniversalAdapterClient {
           throw new Error('connectionControllerClient:signMessage - provider is undefined')
         }
 
-        const signature = await provider.request({
-          method: 'personal_sign',
-          params: [message, address]
-        })
+        let signature = ''
+
+        if (
+          ChainController.state.activeCaipNetwork?.chainNamespace ===
+          CommonConstantsUtil.CHAIN.SOLANA
+        ) {
+          const response = await provider.request(
+            {
+              method: 'solana_signMessage',
+              params: {
+                message: bs58.encode(new TextEncoder().encode(message)),
+                pubkey: address
+              }
+            },
+            ChainController.state.activeCaipNetwork?.caipNetworkId
+          )
+
+          signature = (response as { signature: string }).signature
+        } else {
+          signature = await provider.request(
+            {
+              method: 'personal_sign',
+              params: [message, address]
+            },
+            ChainController.state.activeCaipNetwork?.caipNetworkId
+          )
+        }
```

Comment:
hmmmm won't we need an if for every chain here? 
should we have a unified `signMessage` method that abstract this somehow ?

Reply:
Theoretically this should have been handled by the Solana Adapter, but when Solana adapter is not added as an adapter option it will use UPA and this case was not being handled correctly, that's why I had to add this conditional. If there is different implementation for each network then it UPA will need to handle all of them. I think the way to do it would have these methods implemented somewhere else and then call these methods from this function, what do you think?